### ISSUE TITLE
False positive of `CA1508` on switch expressions

### ISSUE THREAD

**Issue by RenderMichael** [dotnet/roslyn-analyzers#6893]
### Analyzer

**Diagnostic ID**: [CA1508](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**Version**: SDK 8.0 Preview 7

### Describe the bug

```csharp
static bool M(object? value) => value switch
{
    _ => value is not null // CA1508: `value is not null` is always `true`
};
```

## Additional context

No problem with this:
```csharp
static bool M(object? value) => value is not null;
```

**Comment by mavasani** [comment]
Confirmed repro on latest bits.
